src/util/color/colorpalette.cpp

```diff
@@ -38,6 +38,21 @@ mixxx::RgbColor::optional_t ColorPalette::previousColor(mixxx::RgbColor::optiona
     return at(size() - 1);
 }
 
+mixxx::RgbColor::optional_t ColorPalette::getNthColor(
+        mixxx::RgbColor::optional_t color, int steps) const {
+    // TODO : Use rem_euclid modulo function instead of a loop
+    while (steps) {
```

Comment:
we should address this in this PR IMO. The repeated calls to `nextColor`/`previousColor` which in turn call `indexOf` are highly suboptimal (in terms of wasted computations).